tools/upload.py

```diff
@@ -39,28 +39,16 @@
     # https://github.com/esp8266/Arduino/issues/6755#issuecomment-553208688
     if thisarg == "erase_flash":
         write_option = '--erase-all'
-        thisarg = ''
-
-    if thisarg == 'erase_region':
+    elif thisarg == 'erase_region':
         erase_addr = sys.argv.pop(0)
         erase_len = sys.argv.pop(0)
-        thisarg = ''
-
-    # Backwards compatibility with fs upload tools, eat --end
-    if thisarg == '--end':
-        thisarg = ''
-
-    # Backwards compatibility with fs upload tools, parse write_flash for later use
-    if thisarg == 'write_flash':
+    elif thisarg == '--end':
+        # Backwards compatibility with fs upload tools, eat --end
+        pass
+    elif thisarg == 'write_flash':
         write_addr = sys.argv.pop(0)
         binary = sys.argv.pop(0)
-        thisarg = ''
-
-    if os.path.isfile(thisarg):
-        binary = thisarg
-        thisarg = ''
-
-    if len(thisarg):
+    elif len(thisarg):
```

Comment:
Sorry I'm still learning, but should this elif no stay an if? How do we pass the options to cmdline?

Reply:
I mean -chip esp8266 --port...etc.

Reply:
Those parameters won't match any of the other cases, so it will fall through to here.

Reply:
Is this elif not related to :
 if thisarg == "erase_flash":?

Reply:
Ah sorry. Now I got it.